proto/asn1-pdu/mutated_x509_chain_to_der.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include "mutated_x509_chain_to_der.h"
+#include "x509_certificate_to_der.h"
+
+#include <google/protobuf/text_format.h>
+
+#include <stdlib.h>
+#include <utility>
+
+namespace x509_certificate {
+
+// DER encodes each |certificate| in |chain| and returns
+// the encoded certificates in |der|.
+std::vector<std::vector<uint8_t>> EncodeChain(
+    const google::protobuf::RepeatedPtrField<X509Certificate>& chain) {
+  std::vector<std::vector<uint8_t>> der;
+
+  for (const auto& cert : chain) {
+    der.push_back(X509CertificateToDER(cert));
+  }
+
+  return der;
+}
+
+void SetTrust(
+    std::vector<X509>& encoded_mutated_chain,
+    google::protobuf::RepeatedPtrField<TrustParameter>& trust_parameters) {
+  for (const auto& trust_parameter : trust_parameters) {
+    if (trust_parameter.index() >= encoded_mutated_chain.size()) {
+      continue;
+    }
+
+    encoded_mutated_chain[trust_parameter.index()].trusted =
+        trust_parameter.trusted();
+  }
+}
+
+void Mutate(const MutateSignature& mutation,
```

Comment:
```suggestion
void MutateSignature(const MutateSignature& mutation,
```

Combined with my suggestion below, trying to make it explicitly clear in the code that these are different mutations that are happening; that is, trying **not** to use overloads.